docs/Datastore-batchsource.md

```diff
@@ -0,0 +1,90 @@
+# Google Cloud Datastore Source
+
+Description
+-----------
+This source reads data from Google Cloud Datastore.
+Datastore is a NoSQL document database built for automatic scaling and high performance.
+
+Credentials
+-----------
+If the plugin is run on a Google Cloud Dataproc cluster, the service account key does not need to be
+provided and can be set to `auto-detect`. Credentials will be automatically read from the cluster environment.
+
+If the plugin is not run on a Dataproc cluster, the path to a service account key must be provided.
+The service account key can be found on the Dashboard in the Cloud Platform Console.
+Make sure the account key has permission to access Google Cloud Datastore.
+The service account key file needs to be available on every node in your cluster and
+must be readable by all users running the job.
+
+Properties
+----------
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Namespace:** A namespace partitions entities into a subset of datastore. 
+If not provided, `[default]` namespace will be used. 
+
+**Kind:** The kind of an entity categorizes it for the purpose of Datastore queries. 
+Equivalent to relational database table notion.
+
+**Has Ancestor:** Ancestor identifies the common root entity in which the entities are grouped. 
+Must be written in Key Literal format: `key(kind_1, identifier_1, kind_2, identifier_2, [...])`.
+Example: `key(kind_1, 'stringId', kind_2, 100)`.
+
+**Filters:** List of filter property names and values pairs by which equality filter will be applied. 
+This is a semi-colon separated list of key-value pairs, where each pair is separated by a pipe sign `|`. 
+Filter properties must be present in the schema. Allowed property types are STRING, LONG, DOUBLE, BOOLEAN and TIMESTAMP. 
+Property value indicated as `null` string will be treated as `is null` clause. 
+TIMESTAMP string should be in the RFC 3339 format without the timezone offset (always ends in Z). 
+Expected pattern: `yyyy-MM-dd'T'HH:mm:ssX`, example: `2011-10-02T13:12:55Z`. 
```

Comment:
what is the granularity of timestamps in Cloud Datastore? If they are more granular than seconds, this is not useful.

It may be easier to just make users specify the timestamp directly as a number.